extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/common/GoogleErrorLogger.java

```diff
@@ -0,0 +1,26 @@
+package org.datatransferproject.datatransfer.google.common;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.UUID;
+import org.datatransferproject.spi.cloud.storage.JobStore;
+import org.datatransferproject.types.transfer.errors.ErrorDetail;
+
+/**
+ * Helper class that provides functionality for creating errors & logging them to data store.
+ * This class logs errors that should appear as failures in the completion email.
+ */
+public class GoogleErrorLogger {
+  public static void logFailedItemErrors(UUID jobId, ImmutableList<ErrorDetail> errorDetails, JobStore jobStore) throws IOException {
```

Comment:
your call: do we need this method?

Also, should we call it just log() ? it has to be errors. and maybe something other than failed items. So the caller would do

GoogleErrorLogger.log(..); 

which reads ok.

Reply:
Yes I think we do need it, but only because I want to be very specific about what we're logging here. 

This is only to be used when logging an exception that caused an item to fail, because this is then populated in the completion email which informs the user of the results of the transfer. 

If we just call this in-line, I think very easily it could be misused

Reply:
I think we could find better naming here, but GoogleErrorLogger.log is not amazing because people might think this can be used to log anything, when in reality it is only for exceptions that have caused an item to fail.